lib/utils/resolveObject.js

```diff
@@ -100,9 +97,24 @@ function compile_value(value, stack) {
         // Recursive so we can compute multi-layer variables like a = b, b = c, so a = c
         if ( to_ret.indexOf('{') > -1 ) {
           var reference = to_ret.slice(1, -1);
-          if(stack.indexOf(reference)!==-1) {
-            stack.push(reference);
-            throw new Error('Variable reference for \'' + stack[0] + '\' resolves to circular definition cycle of \'' + reference + '\':  ' + stack.join(', ') );
+
+          // Compare to found circular references
+          var foundCircArr = Object.keys(foundCirc);
+          if (foundCircArr.indexOf(reference)!==-1) {
+            // If the current reference is a member of a circular reference, do nothing
+          }
+          else if(stack.indexOf(reference)!==-1) {
+            // If the current stack already contains the current reference, we found a new circular reference
+            // chop down to just the circular part, save it to our circular reference info, and spit out an error
+            var stackIndexReference = stack.indexOf(reference);
+            var circStack = stack.slice(stackIndexReference);
+            circStack.forEach(function(key) { foundCirc[key] = true; });
+            circStack.push(reference);
```

Comment:
TBH I don't really know what is going on here.

Reply:
Adding comments to the code to explain. Essentially creating documentation of the circular reference cycle that we found. Then using that to look for circular references in the future and to output that info to the error system.